EIPS/eip-3540.md

```diff
@@ -262,43 +246,20 @@ The choice of `MAGIC` guarantees that none of the contracts existing on the chai
 
 ## Test Cases
 
-### Contract creation
-
-All cases should be checked for creation transaction, `CREATE` and `CREATE2`.
+### Container validation rules
 
-- Legacy init code
-  - Returns legacy code
-  - Returns valid EOF1 code
-  - Returns invalid EOF1 code, contract creation fails
-  - Returns 0xEF not followed by EOF1 code, contract creation fails
-- Valid EOF1 init code
-  - Returns legacy code, contract creation fails
-  - Returns valid EOF1 code
-  - Returns invalid EOF1 code, contract creation fails
-  - Returns 0xEF not followed by EOF1 code, contract creation fails
-- Invalid EOF1 init code
+(TBD)
 
 ### Contract execution
 
-- EOF code containing `PC` opcode - offset inside code section is returned
-- EOF code containing `CODECOPY/CODESIZE` - works as in legacy code
-  - `CODESIZE` returns the size of entire container
-  - `CODECOPY` can copy from code section
-  - `CODECOPY` can copy from data section
-  - `CODECOPY` can copy from the EOF header
-  - `CODECOPY` can copy entire container
-- `EXTCODECOPY/EXTCODESIZE/EXTCODEHASH` with the EOF *target* contract - works as with legacy target contract
-  - `EXTCODESIZE` returns the size of entire target container
-  - `EXTCODEHASH` returns the hash of entire target container
-  - `EXTCODECOPY` can copy from target's code section
-  - `EXTCODECOPY` can copy from target's data section
-  - `EXTCODECOPY` can copy from target's EOF header
-  - `EXTCODECOPY` can copy entire target container
-  - Results don't differ when executed inside legacy or EOF contract
-- EOF1 `DELEGATECALL`
-  - `DELEGATECALL` to EOF1 code succeeds
-  - `DELEGATECALL` to EOF0 code fails
-  - `DELEGATECALL` to empty container fails
+- EOF0 `EXTCODECOPY` to an EOF1 contract copies 0 bytes
+- EOF0 `EXTCODEHASH` to an EOF1 contract returns hash of `0xEF00`
+- EOF0 `EXTCODESIZE` to an EOF1 contract returns 2
+- EOF0 `DELEGATECALL2` to EOF1 code succeeds
```

Comment:
I think we decided that new calls are only for EOF?
```suggestion
- EOF0 `DELEGATECALL` to EOF1 code succeeds
```

Reply:
correct, I updated too much here